src/rgw/motr/sync/motr_sync_impl.cc

```diff
@@ -29,46 +47,65 @@ int MotrLock::lock(const std::string& lock_name,
   }
   int rc = 0;
   motr_lock_info_t lock_obj;
-  rc = _lock_provider->read_lock(lock_name, &lock_obj);
-  if (rc != 0 && rc != -ENOENT)
-    return rc;
+  // First, try to write lock object
+  struct motr_locker_info_t locker;
+  std::string s_locker_id = locker_id;
+  if (s_locker_id.empty()) {
+    s_locker_id = random_string(32);
+  }
+  locker.cookie = s_locker_id;
+  locker.expiration = ceph_clock_now() + lock_duration;
+  locker.description = "";
+  // Insert lock entry
+  lock_obj.lockers.insert(
+      std::pair<std::string, struct motr_locker_info_t>(locker_id,
+                                                        locker));
+  rc = _lock_provider->write_lock(lock_name, &lock_obj, false);
   if (rc == 0) {
-    // lock entry is present. Check if this is a stale/expired lock
-    utime_t now = ceph_clock_now();
-    auto iter = lock_obj.lockers.begin();
-    while (iter != lock_obj.lockers.end()) {
-      struct motr_locker_info_t &info = iter->second;
-      if (!info.expiration.is_zero() && info.expiration < now) {
-        // locker has expired; delete it
-        iter = lock_obj.lockers.erase(iter);
-      } else {
-        ++iter;
+    // Possiblity that this lock, after creation here, is deleted by
+    // another client. So, try to get lock object (after some millisecs)
+    // to see if it still exists - reconfirm lock.
+    sleep(RECONFIRM_LOCK_TIME_DELAY);
```

Comment:
Hi, I understand the purpose of this sleep and verify, but I am not sure if we should halt the lock function call for the same. 
I have another not-so-good way i.e. client can also check if he still has the lock after processing a chunk of entries from the gc queue. since I mean in the worst case there will be multiple threads processing the same chunk.

Reply:
I agree that, we should not sleep, once we have the lock

Reply:
ok. will remove sleep and the re-check code from here. Will expose another method to do so.

Reply:
Added new interface method "check_lock()" for caller to confirm the lock acquisition.